### PR TITLE
Bugfix/panic with invalid fillers

### PR DESCRIPTION
**Pull request by pinkisemils**
Fixed a panic when a struct was being filled with an incomplete filler i.e. there is a missing or misspelled setting.

### CODE REVIEW

**Comment by pinkisemils** [comment]
I'd like to point out that if the bit of code that coveralls believes is not tested was not executed as part of the added test, the test would fail...

File: filler.go [hunk 1]
```diff
@@ -86,12 +86,30 @@ func (pf *ProviderFiller) fillStruct(structVal reflect.Value, tags map[string]st
...
-		if err := setField(field, settings[tagVal]); err != nil {
-			return err
+		setting, ok := settings[tagVal]
+
+		// If a setting was not found, it will be null.
+		// Since nothing deals with null settings, they can't be passed down.
+		if ok {
+			if err := setField(field, setting); err != nil {
+				return err
+			}
```
Comment (geototti21) [inline filler.go]: you can have a `continue` here so you dont need the else statement. It will make the code cleaner.

Comment (pinkisemils) [inline filler.go]: Would a `continue` not omit the collection of tags there? Those should be accounted for, no? Then again, I guess there already are other early returns.

Comment (geototti21) [inline filler.go]: I think it will not omit them, you only append when you miss a tag.

File: filler_test.go [hunk 2]
```diff
@@ -222,3 +222,50 @@ func TestFiller_MapFillFieldThatDoesNotExist(t *testing.T) {
...
+func TestFiller_SetStringNullCheck(t *testing.T) {
+
+	type someConf struct {
+		User string `setting:"USER"`
+		Pass string `setting:"PASS"`
+	}
+
+	// note the discrepancy between the key for someConf.User
+	var wrongConf = dummySettingsProvider{
+		"USSR": Value("Stalin"),
```
Comment (martinwhitenstall) [inline filler_test.go]: probably cant make these jokes in a public repo

Comment (adamkisala) [inline filler_test.go]: is Stalin a joke for you Martin?

**Comment by pinkisemils** [comment]
Removed the else block and Stalin.

**Pull request merged**